### DISCUSSION TITLE
Plugin API for data about Tailwind classes that were added to CSS file

### DISCUSSION THREAD

**Discussion by dcastil** [tailwindlabs/tailwindcss#10348] [Ideas]
Hi Tailwind people!

I'm the author of [tailwind-merge](https://github.com/dcastil/tailwind-merge), a library that resolves conflicts between Tailwind classes in a string. To be able to do that I rely on a [large config](https://github.com/dcastil/tailwind-merge/blob/v1.8.1/src/lib/default-config.ts) which gets included in the client-side JS bundle and executed at runtime.

I just got a [suggestion](https://github.com/dcastil/tailwind-merge/discussions/185) to prune the output code dynamically like Tailwind's JIT compiler does it to reduce the bundle size impact of tailwind-merge and I would love to do that. For this I'd need to know which Tailwind classes were added to the final `.css` file. My idea was to write a Tailwind plugin which gets data about all the Tailwind classes that were built and then generate code for a matching minimal tailwind-merge config.

As far as I understand there is currently no API for a Tailwind plugin to understand which Tailwind classes got included in the built `.css` file and use that data for further processing. Would you be interested in building an API for that?

Thanks! 😊

**Comment by matthewdavi** [comment]
This would be super helpful. The current runtime size of tailwind-merge puts it on par with certain css-in-js solutions. But it is essential for using Tailwind with any sort of component library like shadcn.

**Comment by aquaductape** [comment]
This would be nice

**Comment by t-mish** [comment]
+1

**Comment by clicktodev** [comment]
+1

**Reply by gregg-cbs** [reply]
+1

**Comment by clicktodev** [comment]
@RobinMalfait @thecrypticace looks like there's plenty of demand for this feature.
Would you mind taking a look and possibly work on an RFC?
Thanks

**Comment by sanex3339** [comment]
Having access to generated classes in some form also allows us to create a minimizers:
- using TW we get the list with all class-names
- based on class-names list, we replace/minimize them in CSS/JS files via e.g. Webpack Plugin

**Comment by dmitrc** [comment]
Is this something that is still being considered, @clicktodev / @RobinMalfait / @thecrypticace?
